src/server/generic_family.cc

```diff
@@ -207,14 +202,11 @@ class RestoreArgs {
 
 [[nodiscard]] bool RestoreArgs::UpdateExpiration(int64_t now_msec) {
   if (HasExpiration()) {
-    auto new_ttl = CalculateExpirationTime(!abs_time_, abs_time_, expiration_, now_msec);
-    if (new_ttl > kMaxExpireDeadlineSec * 1000) {
+    int64_t ttl = abs_time_ ? expiration_ - now_msec : expiration_;
+    if (ttl > kMaxExpireDeadlineSec * 1000)
       return false;
-    }
-    expiration_ = new_ttl;
-    if (new_ttl > 0) {
-      expiration_ += now_msec;
-    }
+
+    expiration_ = ttl < 0 ? -1 : ttl + now_msec;
```

Comment:
Is there a reason to set it to `-1` and not leave it as is ? (again the semantics for `expiration_` can be a expressed in a more modern way, but this is purely stylistic so there is no real value to discuss this here -- I just like to think out loud and that's why I am making this comment)

Reply:
No reason, It just explicitly underlines that negative values have only one purpose - mark an already expired state and the value doesn't matter